src/app/cluster/details/external-cluster/external-machine-deployment-details/template.html

```diff
@@ -38,6 +38,14 @@
                 [disabled]="!isEditEnabled()">
           <i class="km-icon-mask km-icon-edit"></i>
         </button>
+
+        <button mat-icon-button
+                color="tertiary"
+                matTooltip="Delete Machine Deployment"
+                (click)="showDeleteDialog(); $event.stopPropagation()"
```

Comment:
Just for info any harm to place it on? I always thought to stop event propagation when ever there are events so that UP in the chain it shouldn't have any other unwanted effects.

Reply:
By adding this here you might be causing unwanted effects up in the event chain. For example, there could be an open dropdown which won't close due to this stop propagation.